nncf/quantization/algorithms/weight_compression/openvino_backend.py

```diff
@@ -171,55 +261,24 @@ def transform_model(
                         break
 
-            weight = Tensor(get_const_value(const_node))
-            original_shape = weight.shape
-            compressed_weight = compress_weight(
+            mul, compressed_weight = self._create_compression_subgraph(
                 weight,
-                wc_params.reduction_axes,
-                compression_config,
-                None if precomputed_scales is None else precomputed_scales.get(wc_params.weight_name),
-                None if precomputed_zero_points is None else precomputed_zero_points.get(wc_params.weight_name),
+                wc_params,
+                const_node_name,
+                const_dtype,
+                should_add_convert_node,
+                precomputed_scales,
+                precomputed_zero_points,
             )
 
-            compressed_const = opset.constant(
-                compressed_weight.tensor.data, dtype=compression_dtype, name=const_node_name
-            )
-            converted_const = opset.convert(compressed_const, ov.Type.f16)
-            if compressed_weight.zero_point is not None and compressed_weight.tensor.dtype == TensorDataType.uint8:
-                zero_point_const = opset.constant(
-                    compressed_weight.zero_point.data,
-                    dtype=compression_dtype,
-                    name=f"{const_node_name}/zero_point",
-                )
-                converted_zero_point = opset.convert(zero_point_const, ov.Type.f16)
-                converted_const = opset.subtract(
-                    converted_const, converted_zero_point, name=f"{const_node_name}/zero_point/subtract"
-                )
-
-            scale_const = opset.constant(
-                compressed_weight.scale.data, dtype=scale_dtype, name=f"{const_node_name}/scale"
-            )
-            if scale_dtype != ov.Type.f16:
-                scale_const = opset.convert(scale_const, ov.Type.f16)
-
-            mul = opset.multiply(
-                converted_const,
-                scale_const,
-                name=f"{const_node_name}/fq_weights_{wc_params.weight_port_id}",
-            )
-
-            if compression_config.group_size != -1:
-                mul = opset.reshape(mul, output_shape=original_shape, special_zero=False)
-
-            if should_add_convert_node:
-                mul = opset.convert(
-                    mul, const_dtype, name=f"{const_node_name}/fq_weights_{wc_params.weight_port_id}/convert"
-                )
-
             mul_output = mul.output(0)
             for target_input in const_node.output(0).get_target_inputs():
                 target_input.replace_source_output(mul_output)
 
+            if lora_algorithm is not None and lora_algorithm.is_applicable(wc_params):
+                V, US = lora_algorithm.calculate_adapters(weight, compressed_weight, wc_params)
+                self.insert_adapters(wc_params, V, US, int8_lora=lora_algorithm.is_int8_adapters)
```

Comment:
Lora Correction Algorithm is applied layer-by-layer after creating a compressed weight.